packages/@react-spectrum/s2/src/Table.tsx

```diff
@@ -11,85 +11,1089 @@
+  // backgroundColor: '--rowBackgroundColor'
+});
+
+const cellContent = style({
+  truncate: true,
+  whiteSpace: {
+    default: 'nowrap',
+    overflowMode: {
+      wrap: 'normal'
+    }
+  },
+  textAlign: {
+    align: {
+      start: 'start',
+      center: 'center',
+      end: 'end'
+    }
+  },
+  width: 'full'
+});
+
+const cellBackground = style({
+  position: 'absolute',
+  top: 0,
+  bottom: 0,
+  right: 0,
+  left: 0,
+  backgroundColor: {
+    default: 'transparent',
+    isSticky: '--rowBackgroundColor'
+  }
+});
+
+export interface CellProps extends RACCellProps, Pick<ColumnProps, 'align' | 'showDivider'> {
+  /** @private */
+  isSticky?: boolean,
+  /** The content to render as the cell children. */
+  children?: ReactNode
+}
+
+export function Cell(props: CellProps) {
+  let {children, isSticky, showDivider, align, textValue, ...otherProps} = props;
+  let tableVisualOptions = useContext(InternalTableContext);
+  textValue ||= typeof children === 'string' ? children : undefined;
+
+  return (
+    <RACCell
+      // Also isSticky prop is applied just for the layout, will decide what the RAC api should be later
+      // @ts-ignore
+      isSticky={isSticky}
+      // This is a inline style because it needs to set properties ONLY for the end border (don't want to set a color for the bottom border)
+      style={{borderInlineEndColor: showDivider ? 'var(--dividerColor)' : 'none', borderInlineEndWidth: showDivider ? 1 : 0, borderRightStyle: showDivider ? 'solid' : 'none'}}
+      className={renderProps => cell({
+        ...renderProps,
+        ...tableVisualOptions
+      })}
+      textValue={textValue}
+      {...otherProps}>
+      {({isFocusVisible}) => (
+        <>
+          {/*
+            // TODO: problem with having the checkbox cell itself use the row background color directly instead
+            of having a separate white rectangle div base below a div with the row background color set above it as a mask
+            is that it doesn't come out as the same color as the other cells because the base below the sticky cell when other selected cells are scrolled below it will be the blue of the
+            other cells, not the same white base. If I could convert informative-900/10 (and the rest of the rowBackgroundColors) to an equivalent without any opacity
+            then I could do away with this styling. To reproduce this, comment out the stickyCell gray-25, get rid of the below div and apply backgroundColor: '--rowBackgroundColor' to checkboxCellStyle.
+            Having the CellFocusRing here instead of applying a outline on the cell directly also makes it NOT overlap with the border (can be remedied with a -3px outline offset) and applying a border radius to get the curved outline focus ring messes
+            with the divider rendered on the cell since those are also borders
+          */}
+          <div role="presentation" className={cellBackground({isSticky})} />
```

Comment:
Once I get token values for all table row states with full opacity I can do away with this extra element